.changeset/thirty-tomatoes-jump.md

```diff
@@ -0,0 +1,36 @@
+---
+'xstate': minor
+---
+
+The resolved value of the `invoke.data` property is now available in the "invoke meta" object, which is passed as the 3rd argument to the service creator in `options.services`. This will work for all types of invoked services now, including promises, observables, and callbacks. The behavior of `invoke.data` with invoked machines is unchanged.
```

Comment:
It's a little bit nit-picky - but service creators which return machines will from now on be able to use this invoke meta as well. Including this piece of information might make this description too complicated though, so maybe not worth including it? 🤔